### PR TITLE
refactored configuration to a class that lists all the fields

### PR DESCRIPTION
**Pull request by reuvenpo**
The core of the change is in `src/util/config.py`.

### CODE REVIEW

File: src/util/config.py [hunk 1]
```diff
@@ -15,54 +18,99 @@
...
-            if env_name in os.environ:
-                return os.getenv(env_name)
-
-        return super().__getitem__(item)
+__all__ = ['Config', 'config', 'get_config']
+
+
+# The normalizers are pointed out explicitly here for robustness in case the field
+# is written in the config file as a string, or is just fetched from the environment.
+class Config(Model):
```
Comment (Cashmaney) [inline src/util/config.py:26]: Cool use of serde and all, but is there any reason to use this over a dataclass?

**Comment by Cashmaney** [comment]
Also why update the contracts?

File: src/util/config.py (hunk 1 above)
Comment (reuvenpo) [reply src/util/config.py:26]: The `from_dict` method used in the end of `get_config` lets me easily convert from a primitive type to an instance of this class. Plus you can call `to_json` if you want to print the full configuration for some reason.

**Comment by reuvenpo** [comment]
>Also why update the contracts?

Now that i think of it i probably didn't need to. The tests started failing on master, and i was trying to fix them at first. i saw that there were panics coming out of the swap contract initialization, so i tried rebuilding it at first.

File: src/util/config.py (hunk 1 above)
Comment (Cashmaney) [reply src/util/config.py:26]: It doesn't really matter here, but in general I would prefer using objects from the standard library, and if you need to create it from a dict, there's probably a small library that does that (also, dataclasses have `asdict` if you want to print it).

Even better would be to just use a standard configuration library, but whatever, not worth the effort

File: src/util/config.py [hunk 2]
```diff
@@ -15,54 +18,99 @@
...
+    except IOError:
+        logger.critical("there was a problem opening the config file")
+        raise
+    except json.JSONDecodeError as e:
+        logger.critical("config file isn't valid json")
+        raise ValueError from e
+
+    config_data = {}
+    for field_name, field_type in Config.__fields__.items():
+        for source in [os.environ, conf_file_data]:
```
Comment (Cashmaney) [inline src/util/config.py:100]: Which one takes priority?

Comment (reuvenpo) [reply src/util/config.py:101]: the first one, `os.environ`. as soon as a match is found, a `break` is executed

File: src/util/config.py (hunk 1 above)
Comment (reuvenpo) [reply src/util/config.py:26]: I'm not up to speed with all the new python3 packages tbh. is there a decent standard package with this functionality?

Comment (Cashmaney) [reply src/util/config.py:25]: I meant like the most common libraries for configuration management, not necessarily `configparser`:)

**Pull request merged**